### ISSUE TITLE
Metatensor can modify values？

### ISSUE THREAD

**Issue by hz1z** [Project-MONAI/MONAI#7935]
In a segmentation task, I intended to change the regions labeled as -1 to background labeled as 0. I wrote a transform for this purpose, but it did not work 
```python
class SetValued(Transform):
    def __init__(self,keys):
        self.keys = [keys]
    def __call__(self, inputs):
        for key in self.keys:
            for c in range(inputs[key].shape[0]):
                inputs[key][c][inputs[key][c] == -1] = 0
        return inputs
```
This transform is located in…
![image](https://github.com/user-attachments/assets/c7248ba1-6ca5-4da2-9a0a-b4f223e92b46)

But when I debugged, I found that my modifications had no effect.
![image](https://github.com/user-attachments/assets/a7bc83fb-05c1-4d63-b2ec-7306168cf610)

**Comment by mingxin-zheng** [comment]
Hi @hz1z , a short question, can you try the below line and see if it returns `True`?
```
torch.min(inputs[key][c]) == -1
```

The reason I asked is because the `Resized` transform can alter values in label, and introduce small error to the integers in label, making them floats (1.00000 +/- epsilon).

The following snippets works fine for what you asked:
```
import torch
from monai.transforms import Transform, Compose
from monai.data import MetaTensor

class SetValued(Transform):
    def __init__(self, keys):
        self.keys = keys

    def __call__(self, inputs):
        for key in self.keys:
            # Iterate over each channel in the tensor
            for c in range(inputs[key].shape[0]):
                # Set regions with value -1 to 0
                inputs[key][c][inputs[key][c] == -1] = 0
        return inputs

# Simulate the input data with a label tensor containing -1
input_data = {
    "image": MetaTensor([[[-1.0, 1.0], [0.0, -1.0]]], dtype=torch.float32), # Dummy image
    "label": MetaTensor([[[-1, 1], [0, -1]]], dtype=torch.int8)  # # Dummy image
}

# Define the transformation
train_transforms = Compose([
    SetValued(keys=["image", "label"])
])

# Apply the transformation
transformed_data = train_transforms(input_data)

# Verify the result
print("Transformed images:\n", transformed_data["image"])
print("Transformed labels:\n", transformed_data["label"])
```

**Comment by hz1z** [comment]
![image](https://github.com/user-attachments/assets/c4bce6d6-ace7-4f4b-863e-29b4dd04eb0f)
Thank you for your reply.  torch(inputs[key][c])== -1 is indeed not true, torch.min(inputs[key][c]) is -1.0000. However, even after changing -1 to -1.0000, it still doesn’t work. Is there perhaps an issue with metatensor when it comes to comparing numerical values?
-----
Additionally, I’ve tried many ways in the VSCode debug console, such as:
![image](https://github.com/user-attachments/assets/66764f0f-34d3-490e-a5fb-6094e012b244)

![image](https://github.com/user-attachments/assets/72c1e267-bd7e-4c04-a061-b27d56a64f33)

**Comment by KumoLiu** [comment]
Hi @hz1z, 
To avoid the issue caused by precision, you may want to convert the label to int before setting it by using
```
EnsureTyped(keys="label", dtype=torch.int16)
```
https://github.com/Project-MONAI/MONAI/blob/316934aba4093ff4657140819c8797177b2b0259/monai/transforms/utility/dictionary.py#L472

And I noticed that you are using `linear` mode in `Resize` for the label, I guess that's not your intention, because that creates decimals, and you should use `nearest` mode for the label.

BTW, I would recommend you use `Lambdad` instead of writing a new transform.
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/utility/dictionary.py#L980

**Comment by hz1z** [comment]
than you! @KumoLiu  @mingxin-zheng 
it's work after use ```nearest``` mode!!!

![image](https://github.com/user-attachments/assets/615f8366-0e3d-41d5-a3a9-448e5b94e6c9)
